packages/material-ui/src/ButtonBase/TouchRipple.test.js

```diff
@@ -1,88 +1,114 @@
-import { createShallow, createMount, getClasses, unwrap } from '@material-ui/core/test-utils';
-import describeConformance from '../test-utils/describeConformance';
-import Ripple from './Ripple';
+import { expect } from 'chai';
+import { createMount, getClasses } from '@material-ui/core/test-utils';
+import { cleanup, createClientRender } from 'test/utils/createClientRender';
 import TouchRipple, { DELAY_RIPPLE } from './TouchRipple';
 
 const cb = () => {};
 
 describe('<TouchRipple />', () => {
-  let shallow;
   let mount;
-  let classes;
-  const TouchRippleNaked = unwrap(TouchRipple);
+  // StrictModeViolation: uses react-transition-group
+  const render = createClientRender({ strict: false });
+
+  /**
+   * @param {HTMLElement} container returned by render
+   * @param {number} active of active ripples to check for
+   * @param {number} stopping of stopping ripples to check for
+   */
+  function checkRipples(container, active, stopping) {
+    expect(
+      container.querySelectorAll('.ripple-visible .child:not(.child-leaving)'),
+    ).to.have.lengthOf(active);
+    expect(container.querySelectorAll('.ripple-visible .child-leaving')).to.have.length(stopping);
+  }
+
+  /**
+   * @param {object} other props to spread to TouchRipple
+   */
+  function createTouchRipple(other) {
+    const touchRippleRef = React.createRef();
+    const { container } = render(
+      <TouchRipple
+        ref={touchRippleRef}
+        classes={{
+          ripple: 'ripple',
+          rippleVisible: 'ripple-visible',
+          child: 'child',
+          childLeaving: 'child-leaving',
+        }}
+        {...other}
+      />,
+    );
+
+    const instance = touchRippleRef.current;
+    return { container, instance };
+  }
 
   before(() => {
-    shallow = createShallow({ dive: true });
-    mount = createMount({ strict: undefined });
-    classes = getClasses(<TouchRipple />);
+    // StrictModeViolation: uses react-transition-group
+    mount = createMount({ strict: false });
   });
 
   after(() => {
+    cleanup();
     mount.cleanUp();
   });
 
+/* Breaks due to ref being used for an imperative handle
   describeConformance(<TouchRipple />, () => ({
     classes,
     inheritComponent: 'span',
     mount,
     refInstanceof: React.Component,
     skip: ['componentProp'],
-  }));
+  })); */
 
   describe('prop: center', () => {
     it('should should compute the right ripple dimensions', () => {
-      const wrapper = mount(<TouchRipple center />);
-      const instance = wrapper.find('TouchRipple').instance();
+      const { instance, container } = createTouchRipple({ center: true });
+
       instance.start(
         {},
         {
           fakeElement: true,
         },
         cb,
       );
-      wrapper.update();
-      assert.strictEqual(
-        wrapper
-          .find(Ripple)
-          .at(0)
-          .props().rippleSize,
-        1,
-      );
+      expect(container.querySelector('.ripple').style).to.have.property('height', '1px');
```

Comment:
I love the new testing strategy ❤️ 

@eps1lon

Reply:
Me too. I actually have fun writing tests. 

If only jsdom would fully implement style then we could do `expect(container.querySelector('.ripple')).to.have.computedStyle('height', '1px')`. But jsdom's computed style doesn't walk the full DOM tree (not to be confused with the missing layout engine) so this would give a false sense of confidence.

Reply:
Understandable, I was just trying to read the width/height directly applied